backend/apps/investors/permissions.py

```diff
@@ -14,4 +15,24 @@ class IsInvestorSelf(permissions.BasePermission):
     matches the authenticated user's ID.
     """
     def has_permission(self, request, view):
-        return request.user.id == view.kwargs.get('investor_id')
\ No newline at end of file
+        return request.user.id == view.kwargs.get('investor_id')
```

Comment:
IsInvestorSelf checks view.kwargs.get('investor_id'), while InvestorInvestmentsListView uses id as the kwarg in the route. This inconsistency can result in permissions incorrectly denying access (or KeyErrors if code expects a value).
Make the URL kwarg name consistent (prefer investor_id) and update all references/tests to match; or read both keys defensively:
```
investor_id = view.kwargs.get('investor_id') or view.kwargs.get('id')
return request.user.id == investor_id
```